src/NRediSearch/Client.cs

```diff
@@ -50,19 +50,104 @@ public enum IndexOptions
             KeepTermFrequencies = 16
         }
 
+        public sealed class IndexDefinition
+        {
+            public enum IndexType
+            {
+                /// <summary>
+                /// Used to indicates that the index should follow the keys of type Hash changes
+                /// </summary>
+                HASH
+            }
+
+            internal readonly IndexType _type = IndexType.HASH;
+            internal readonly bool _async; 
+            internal readonly string[] _prefixes;
+            internal readonly string _filter;
+            internal readonly string _languageField;
+            internal readonly string _language;
+            internal readonly string _scoreFiled;
+            internal readonly double _score;
+            internal readonly string _payloadField;
+
+            public IndexDefinition(bool async = false, string[] prefixes = null,
+            string filter = null, string languageField = null, string language = null, 
+            string scoreFiled = null, double score = 1.0, string payloadField = null)
+            {
+                _async = async;
+                _prefixes = prefixes;
+                _filter = filter;
+                _languageField = languageField;
+                _language = language;
+                _scoreFiled = scoreFiled;
+                _score = score;
+                _payloadField = payloadField;
+            }
+
+            internal void SerializeRedisArgs(List<object> args)
+            {
+                args.Add("ON".Literal());
+                args.Add(_type.ToString("g"));
```

Comment:
to handle both casing and the unknown enums, consider:

``` c#
args.Add(_type switch {
    IndexType.Hash => "HASH".Literal(),
    _ => throw new InvalidOperationException($"Unexpected index kind: {_type}")
});
```

Reply:
There is no need to handle casing, both `Hash` and `HASH` will work.